pkgs/by-name/zi/ziti/package.nix

```
@@ -0,0 +1,41 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule (finalAttrs: {
+  pname = "ziti";
+  version = "1.5.4";
+
+  src = fetchFromGitHub {
+    owner = "openziti";
+    repo = "ziti";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-u8OMnSUKlOiQtPtJLQrUX6M5IddDTds/77RHcAyhWeQ=";
+  };
+
+  vendorHash = "sha256-P9+trbU6lLv/jMXojI+BwQovsrBwkO+vr/pdJgdx0wo=";
+
+  subPackages = [
+    "ziti"
+    "controller"
+    "router"
+  ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/openziti/ziti/common/version.Version=v${finalAttrs.version}"
+    "-X github.com/openziti/ziti/common/version.Revision=v${finalAttrs.version}"
+  ];
+
```

I think we can add `versionCheckHook` to test if the binary is running fine:

```suggestion
  doInstallCheck = true;
  nativeInstallCheckInputs = [ versionCheckHook ];
  versionCheckProgramArg = "version";
```

What do you think?